test/e2e/common_suite_test.go

```diff
@@ -114,15 +120,58 @@ func (tc *testCase) run() {
 				if err = client.Resources().Create(ctx, tc.pod); err != nil {
 					t.Fatal(err)
 				}
-				if err = wait.For(conditions.New(client.Resources()).PodRunning(tc.pod), wait.WithTimeout(WAIT_POD_RUNNING_TIMEOUT)); err != nil {
-					t.Fatal(err)
+				if tc.podState != "" {
+					if err = wait.For(conditions.New(client.Resources()).PodPhaseMatch(tc.pod, v1.PodPhase(tc.podState)), wait.WithTimeout(WAIT_POD_RUNNING_TIMEOUT)); err != nil {
+						t.Fatal(err)
+					}
+				} else {
+					if err = wait.For(conditions.New(client.Resources()).PodRunning(tc.pod), wait.WithTimeout(WAIT_POD_RUNNING_TIMEOUT)); err != nil {
+						t.Fatal(err)
+					}
 				}
+
 			}
 			return ctx
 		}).
 		Assess(tc.assessMessage, func(ctx context.Context, t *testing.T, cfg *envconf.Config) context.Context {
 			client := cfg.Client()
 			var podlist v1.PodList
+			if tc.pod != nil {
+				if err := client.Resources(tc.pod.Namespace).List(ctx, &podlist); err != nil {
+					t.Fatal(err)
+				}
+				if tc.expectedPodLogString != "" {
+					isLogStringValid, LogString, err := comparePodLogString(ctx, t, client, *tc.pod, tc.expectedPodLogString)
+					if err != nil {
+						t.Fatal(err)
+					}
+					if isLogStringValid {
+						t.Logf("Log output of peer pod:%s", LogString)
+					} else {
+						t.Errorf("Invalid Log output from peer pod:%s", LogString)
+					}
+				} else if tc.testCommands != nil {
```

Comment:
I don't think having checking logs and running a command inside a pod container in exclusive branches is going to be valid for all test cases we could write, because you could write a test case where we want to check the logs in a pod that reaches the Running phase, and then run a command inside a container in that running pod.

I suggest checking if `tc.podState` is `v1.PodRunning` and running the test commands if that condition is true, and making that conditional separate from the `tc.expectedPodLogString != ""` check.